workflow/controller/workflowpod.go

```diff
@@ -488,6 +488,12 @@ func addSchedulingConstraints(pod *apiv1.Pod, wfSpec *wfv1.WorkflowSpec, tmpl *w
 	} else if wfSpec.SchedulerName != "" {
 		pod.Spec.SchedulerName = wfSpec.SchedulerName
 	}
+	// set hostaliases
+	if len(tmpl.HostAliases) > 0 {
+		pod.Spec.HostAliases = append(pod.Spec.HostAliases, tmpl.HostAliases...)
+	} else if len(wfSpec.HostAliases) > 0 {
```

Comment:
Since host aliases are a list, they can be merged.

Instead of an if/else, I think we can do an append. So the behavior would be template level host aliases will augment any workflow level host aliases, instead of replacing them.